hyperspy/io.py

```diff
@@ -812,12 +812,13 @@ def dict2signal(signal_dict, lazy=False):
                     value = function(value)
                 if value is not None:
                     signal.metadata.set_item(mpattr, value)
-    if "metadata" in signal_dict and "Markers" in mp:
-        markers_dict = markers_metadata_dict_to_markers(
+    if mp is not None and "Markers" in mp:
+        if "Markers" in signal.metadata:
+            del signal.metadata.Markers
```

Comment:
```suggestion
```
What's about removing it completely? It is very very rare that we use `del` before an assignment in python and I don't see why it is necessary here. Looking at the history using git blame, it doesn't seem that there is a reason.